cnd/src/swap_digest.rs

```diff
@@ -0,0 +1,30 @@
+use multihash::Multihash;
+
+enum Error {
+    SerdeCbor(()),
+}
+
+trait Digest {
+    fn digest(&self) -> Multihash;
+}
+
+impl Digest for String {
```

Comment:
I'm happy with this name, `Digest` is a noun and this function returns the digest and _does not_ change program state.  (Comment added here as preamble to discussion on names of `DigestField` and `DigestRoot`.)